xtable-utilities/src/main/java/org/apache/xtable/utilities/RunSync.java

```diff
@@ -115,66 +117,108 @@ public class RunSync {
               "The interval in seconds to schedule the loop. Requires --continuousMode to be set. Defaults to 5 seconds.")
           .addOption(HELP_OPTION, "help", false, "Displays help information to run this utility");
 
-  public static void main(String[] args) throws IOException {
-    CommandLineParser parser = new DefaultParser();
+  static SourceTable sourceTableBuilder(
+          @NonNull DatasetConfig.Table table,
+      IcebergCatalogConfig icebergCatalogConfig,
+          @NonNull DatasetConfig datasetConfig,
+      Properties sourceProperties) {
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(table.getTableName())
+            .basePath(table.getTableBasePath())
+            .namespace(table.getNamespace() == null ? null : table.getNamespace().split("\\."))
+            .dataPath(table.getTableDataPath())
+            .catalogConfig(icebergCatalogConfig)
+            .additionalProperties(sourceProperties)
+            .formatName(datasetConfig.sourceFormat)
+            .build();
+    return sourceTable;
+  }
 
-    CommandLine cmd;
-    try {
-      cmd = parser.parse(OPTIONS, args);
-    } catch (ParseException e) {
-      new HelpFormatter().printHelp("xtable.jar", OPTIONS, true);
-      return;
-    }
+  static List<TargetTable> targetTableBuilder(
+          @NonNull DatasetConfig.Table table,
+          IcebergCatalogConfig icebergCatalogConfig,
```

Comment:
`IcebergCatalogConfig icebergCatalogConfig` change this to `CatalogConfig catalogConfig`

Reply:
@vinishjail97 I have done the required changes in the functions. Please review.